src/plugins/vis_type_vega/public/data_model/types.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AggParamsMapping, Filter } from 'src/plugins/data/public';
+import { DslQuery } from 'src/plugins/data/common';
+
+interface Coordinate {
+  axis: {
+    title: string;
+  };
+  field: string;
+}
+
+interface Encoding {
+  x: Coordinate;
+  y: Coordinate;
+}
+
+interface AutoSize {
+  type: string;
+  contains: string;
+}
+
+export interface VegaSpec {
```

Comment:
There are still quite some `any`s in here I would like to get rid of.
* Can we somehow leverage types from the vega lib itself to help us? If not, the current approach of typing what we use is fine for me
* For stuff we don't know, I would like to keep them as `unknown` - if that doesn't work, we should use the existing code to infer what type they actually are.

Reply:
👍